function parse

DEDMOPOZ opened the pull request:
Реализация функцию "parse"

DEDMOPOZ committed:
function parse

romanitalian commented on main.py:
```diff
@@ -1 +1,10 @@
+def parse(query: str) -> dict:
+    return {}
```
Тело функции не реализовано.
Что будет если запустить этот файл ?)

DEDMOPOZ committed:
new

DEDMOPOZ committed:
parce_function_second

romanitalian commented on main.py:
```diff
@@ -1 +1,21 @@
+            list = (query.split("?")[1]).split("&")
+        except IndexError:
+            return {}
+        tmp_dict = {}
```
```

        tmp_dict = {}
        try:
            list = (query.split("?")[1]).split("&")
        except IndexError:
            return {}
        tmp_dict = {}
```
Здесь у нас два дефолтных значения: 
- return {}
- tmp_dict = {}

лучше вынести выше
```
    tmp_dict = {}
    try:
        list = (query.split("?")[1]).split("&")
    except IndexError:
        return tmp_dict
```

romanitalian commented on main.py:
```diff
@@ -1 +1,21 @@
+            except IndexError:
+                break
 
+        return tmp_dict
```
А вообще, можно воспользоваться библиотекой:
`from urllib import parse`
Если есть желание - попробуй этот вариант)

romanitalian commented on main.py:
```diff
@@ -1 +1,21 @@
+    assert parse('https://example.com/path/to/page?name=ferret&color=purple&') == {'name': 'ferret', 'color': 'purple'}
+    assert parse('http://example.com/') == {}
+    assert parse('http://example.com/?') == {}
+    assert parse('http://example.com/?name=Dima') == {'name': 'Dima'}
```
И файл должен заканчиваться пустой строкой.
`PEP 8: W292 no newline at end of file`

DEDMOPOZ committed:
Update main.py

Внес изменения согласно требованиям. Использовать urllib попробую в другом бранче
